tower-balance/src/pool/mod.rs

```diff
@@ -61,6 +62,24 @@ where
     limit: Option<usize>,
 }
 
+impl<MS, Target, Request> fmt::Debug for PoolDiscoverer<MS, Target, Request>
+where
+    MS: MakeService<Target, Request> + fmt::Debug,
+    MS::Future: fmt::Debug,
```

Comment:
why does this need to be debug?

Reply:
Because `PoolDiscover` internally holds a `MS::Future`, which is printed out for `Debug`.

Reply:
That seems incorrect though, I would assume that it wouldn't need to print the future? Not sure though.

Reply:
So, the alternative here is https://github.com/tower-rs/tower/pull/352/commits/8c8dddf5fa575019ff33970742734f272ef6a751. Do you prefer it that way?

Reply:
Yeah, I think that looks great! Thanks <3